BasiliskII/src/SDL/audio_sdl.cpp

```diff
@@ -215,10 +219,16 @@ static void stream_func(void *arg, uint8 *stream, int stream_len)
 				goto silence;
 
 			// Send data to audio device
-			Mac2Host_memcpy(stream, ReadMacInt32(apple_stream_info + scd_buffer), work_size);
-			if (work_size != stream_len)
-				memset((uint8 *)stream + work_size, silence_byte, stream_len - work_size);
+			Mac2Host_memcpy(audio_mix_buf, ReadMacInt32(apple_stream_info + scd_buffer), work_size);
```

Comment:
`audio_mix_buf` is allocated with a size of `audio_spec.size`, which is the size that `SDL_OpenAudio` calculates for the audio buffer in bytes.  See [the docs for `SDL_AudioSpec`](https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlaudiospec.html). It would be very weird for our `stream_func` to get an audio buffer bigger than what SDL told us.